Casks/p/paraview.rb

```diff
@@ -1,20 +1,21 @@
 cask "paraview" do
   arch arm: "arm64", intel: "x86_64"
-  min_macos_version = on_arch_conditional arm: "11.0", intel: "10.13"
+  min_macos_version = on_arch_conditional arm: "11.0", intel: "10.15"
 
-  version "5.11.2"
-  sha256 arm:   "0fc82a819996b80014df24e96095b87f9ceae0f65a0c2bf8157aa2ddda4baa0b",
-         intel: "f225784e0b4fa7677b7fc2a4bf7860e9842a032792fe696479f84e4783783220"
+  version "5.12.0"
+  sha256 arm:   "6b50e2d90fd2d2caf1792a66358a854f59aa9c4825b3a8ec997339c893c94bb6",
+         intel: "1b9aca43835a58fb490137031434e4fcbdd4a07247038e3931624a0139917f4b"
 
-  url "https://www.paraview.org/paraview-downloads/download.php?submit=Download&version=v#{version.major_minor}&type=binary&os=macOS&downloadFile=ParaView-#{version}-MPI-OSX#{min_macos_version}-Python3.9-#{arch}.dmg",
+  url "https://www.paraview.org/paraview-downloads/download.php?submit=Download&version=v#{version.major_minor}&type=binary&os=macOS&downloadFile=ParaView-#{version}-MPI-OSX#{min_macos_version}-Python3.10-#{arch}.dmg",
       user_agent: :fake
   name "ParaView"
   desc "Data analysis and visualization application"
   homepage "https://www.paraview.org/"
 
   livecheck do
     url "https://www.paraview.org/files/listing.txt"
-    regex(/ParaView[._-](\d+(?:\.\d+)+)[._-]MPI[._-]OSX#{min_macos_version}[._-]Python3\.9[._-]#{arch}\.dmg/i)
+    regex(%r{(?<!nightly/)ParaView[._-](\d+(?:\.\d+)+)[._-]MPI[._-]OSX#{min_macos_version}[._-]Python3\.10[._-]#{arch}
+          \.dmg}ix)
```

Comment:
```suggestion
    regex(%r{/v?(\d+(?:\.\d+)+)/ParaView[._-]v?(\d+(?:\.\d+)+).*?[._-]#{arch}\.dmg}i)
```

Looking at `listing.txt`, we can omit unstable versions by matching a `/v?(\d+(?:\.\d+)+)` prefix (so we don't have to use a lookaround). That will omit files in directories like `/v5.11/RCs/` and `/nightly/`.

Besides that, we can/should simplify this regex, so it will surface new versions regardless of the macOS and Python versions. We only need to include `min_macos_version` in the regex if there's a different version for ARM and Intel, which doesn't appear to be the case. Similarly, they're not publishing multiple files with different Python versions, so there isn't any benefit to matching a specific Python version. The regex predates this PR but I only mention this so we clean this up while we're looking at it.

Reply:
> Looking at `listing.txt`, we can omit unstable versions by matching a `/v?(\d+(?:\.\d+)+)` prefix (so we don't have to use a lookaround). That will omit files in directories like `/v5.11/RCs/` and `/nightly/`.
> 
> Besides that, we can/should simplify this regex, so it will surface new versions regardless of the macOS and Python versions. We only need to include `min_macos_version` in the regex if there's a different version for ARM and Intel, which doesn't appear to be the case. Similarly, they're not publishing multiple files with different Python versions, so there isn't any benefit to matching a specific Python version. The regex predates this PR but I only mention it so we clean this up while we're looking at it.

So since they really only offer one version for ARM and one version for Intel, should we remove the `min_macos_version` key from the first stanza altogether?

Reply:
I was thinking the same thing. We usually use a comma-separated `version` for this type of scenario and the current setup requires manual edits whenever the macOS or Python version in the filename changes. Using something like `5.12.0,11.0,3.10` or `5.12.0,MPI-OSX-11.0-Python3.10` is more in line with other casks. The former may periodically require manual edits if the suffix changes (it has in the past) whereas the latter would work handle any suffix changes before the `arch`.

Do we have a preference for one or the other here, @Homebrew/cask?

Reply:
I would suggest the latter @samford to minimise the need for manual edits, thanks.

Reply:
I pushed a commit to update the `version` to use a `5.12.0,MPI-OSX-11.0-Python3.10` format (and modified the `url` and `livecheck` block accordingly). All of the existing dmg file names have text between the version and architecture suffix but I made it optional in the `livecheck` block regex (for the sake of flexibility), so the `url` follows suit.

Reply:
Thanks @samford.